src/lib/lib.cpp

```diff
@@ -166,7 +166,7 @@ RHVoice_tts_engine_struct::RHVoice_tts_engine_struct(const RHVoice_init_params*
       const char** p=init_params->resource_paths;
       while(*p)
         {
-          engine_params.resource_paths.push_back(std::string(*p));
+          engine_params.resource_paths.emplace_back(*p);
```

Comment:
not sure if I like this one for [code readability](https://abseil.io/tips/112)
reasons

Reply:
@alex19EP,
U have a pointer declared in the constructor `std::string`, using the employment method you can reduce the size of the code, and there is not much to optimize.
U attached a link to the material that described the principle of the method very well, it is simply convenient and helps the compiler to choose the constructor move and copy for your object.

```cpp
void f2(const std::vector<std::string>& users) {
    std::vector<std::pair<std::string, int>> userOccurancies(users.size());

    auto userOcIt = userOccurancies.begin();
    for (const auto & user : users) {
        userOcIt->first = std::move(user);
        userOcIt->second = 0;
        userOcIt++;
    }
}

void f(const std::vector<std::string>& users) {
    std::vector < std::pair<std::string, std::size_t>> userCount;
    userCount.reserve(users.size());

    for (auto& user : users) {
        userCount.emplace_back(user, 0);
    }
}
```

above wrote an example code to better understand f2() function identical to f()